internal/pkg/cli/local_run.go

```diff
@@ -106,11 +196,195 @@ func (o *localRunOpts) Execute() error {
+	decryptedSecrets, err := o.ecsLocalClient.DecryptedSecrets(secrets)
 	if err != nil {
 		return fmt.Errorf("get secret values: %w", err)
 	}
 
+	envVars := make(map[string]string)
+	envVariables := taskDef.EnvironmentVariables()
+	for _, envVariable := range envVariables {
+		envVars[envVariable.Name] = envVariable.Value
+	}
+
+	containerPorts := make(map[string]string)
+	containerdef := taskDef.ContainerDefinitions
+	for _, container := range containerdef {
+		for _, portMapping := range container.PortMappings {
+			hostPort := aws.Int64Value(portMapping.HostPort)
+			hostPortStr := strconv.FormatInt(hostPort, 10)
+			containerport := aws.Int64Value(portMapping.ContainerPort)
+			containerportStr := strconv.FormatInt(containerport, 10)
+			containerPorts[hostPortStr] = containerportStr
+		}
+	}
+
+	env, err := o.store.GetEnvironment(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("get environment %q configuration: %w", o.envName, err)
+	}
+	o.targetEnv = env
+
+	app, err := o.store.GetApplication(o.appName)
+	if err != nil {
+		return fmt.Errorf("get application %q configuration: %w", o.appName, err)
+	}
+	o.targetApp = app
+
+	envSess, err := o.sessProvider.FromRole(env.ManagerRoleARN, env.Region)
+	if err != nil {
+		return fmt.Errorf("get env session: %w", err)
+	}
+	mft, err := workloadManifest(&workloadManifestInput{
+		name:         o.wkldName,
+		appName:      o.appName,
+		envName:      o.envName,
+		interpolator: o.newInterpolator(o.appName, o.envName),
+		ws:           o.ws,
+		unmarshal:    o.unmarshal,
+		sess:         envSess,
+	})
+	if err != nil {
+		return err
+	}
+	o.appliedDynamicMft = mft
+	o.repository, err = o.configureClients(o)
+	if err != nil {
+		return err
+	}
+
+	err = o.buildContainerImages(o)
+	if err != nil {
+		return fmt.Errorf("building container image: %w", err)
+	}
+
+	containerNames := o.out.ContainerNames
+	imageNames := o.out.ImageNames
+
+	secretsList := make(map[string]string)
+	for _, s := range decryptedSecrets {
+		secretsList[s.Name] = s.Value
+	}
+
+	for i, image := range imageNames {
+		imageInfo := imageInfo{
+			containerName: containerNames[i],
+			imageURI:      image,
+		}
+		o.imageInfoList = append(o.imageInfoList, imageInfo)
+	}
+
+	var sideCarListInfo []imageInfo
+	manifestContent := o.appliedDynamicMft.Manifest()
+	switch t := manifestContent.(type) {
+	case *manifest.ScheduledJob:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.LoadBalancedWebService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.WorkerService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.BackendService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	}
+	o.imageInfoList = append(o.imageInfoList, sideCarListInfo...)
+
+	err = o.runPauseContainer(containerPorts)
+	if err != nil {
+		return err
+	}
+
+	err = o.runContainers(o.imageInfoList, secretsList, envVars)
+	if err != nil {
+		return err
+	}
```

Comment:
i'm wondering if we should prefix or suffix the container name with a random identifier to create run "sessions". that would enable someone to use `local run` for multiple services without running into naming conflicts

Reply:
or maybe it just makes sense to prefix all the containers (including pause) with the app, env, and service name?

Reply:
Thats a good idea.